dagobah/core/delegator.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+from ..backend.base import BaseBackend
+
+logger = logging.getLogger('dagobah')
+
+class CommitDelegator(object):
+    """
+    The CommitDelegator is a class designed to handle any and all commits
+    to the database.
+
+    The delegator takes in commit requests from various objects, and decides
+    if they should actually be committed, and if so, manage any special
+    processing.
+    """
+
+    def __init__(self, backend=BaseBackend()):
```

Comment:
Good point, I think I modeled this off of the existing default behavior you mentioned, but should probably be no default.